src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -151,12 +189,19 @@ static constexpr size_t kPersistentFabricBufferMax = 128;
  */
 struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
 {
-    FabricIndex fabric_index = kUndefinedFabricIndex;
-    uint8_t scene_count      = 0;
+    FabricIndex fabric_index;
+    uint8_t scene_count = 0;
+    uint8_t max_scenes_per_fabric;
+    uint8_t max_scenes_global;
     SceneStorageId scene_map[kMaxScenesPerFabric];
 
-    FabricSceneData() = default;
-    FabricSceneData(FabricIndex fabric) : fabric_index(fabric) {}
+    FabricSceneData(FabricIndex fabric = kUndefinedFabricIndex, uint8_t maxScenePerFabric = kMaxScenesPerFabric,
+                    uint8_t maxScenesGlobal = kMaxScenesGlobal) :
+        fabric_index(fabric)
+    {
+        max_scenes_per_fabric = (maxScenePerFabric > kMaxScenesPerFabric) ? kMaxScenesPerFabric : maxScenePerFabric;
```

Comment:
should prevent users from providing values > than kMax* instead of overriding their values?

Reply:
That is indeed better, now checks the sizes are respected in the Init() of DefaultSceneTableImpl